core-bundle/src/Resources/contao/library/Contao/Dbafs.php

```diff
@@ -84,8 +84,8 @@ public static function addResource($strResource, $blnUpdateFolders=true)
 		}
 
 		$arrPaths    = array();
-		$arrChunks   = explode('/', $strResource);
-		$strPath     = array_shift($arrChunks);
+		$arrChunks   = array_filter(explode('/', Path::makeRelative($strResource, $uploadPath)));
+		$strPath     = $uploadPath;
```

Comment:
Are you sure that this is correct? Not only might `array_shift($arrChunks)` return something else, but now the first chunk is kept in `$arrChunks`, which will affect the while loop in line 94, won‘t it?